tests/python/unittest/test_random.py

```diff
@@ -893,14 +893,15 @@ def compute_expected_prob():
 def test_shuffle():
     def check_first_axis_shuffle(arr):
         stride = int(arr.size / arr.shape[0])
-        column0 = arr.reshape((arr.size,))[::stride].sort()
+        column0 = arr.reshape((arr.size,))[::stride]
         seq = mx.nd.arange(0, arr.size - stride + 1, stride, ctx=arr.context)
-        assert (column0 == seq).prod() == 1
-        for i in range(arr.shape[0]):
-            subarr = arr[i].reshape((arr[i].size,))
-            start = subarr[0].asscalar()
-            seq = mx.nd.arange(start, start + stride, ctx=arr.context)
-            assert (subarr == seq).prod() == 1
+        assert (column0.sort() == seq).prod() == 1
```

Comment:
curious to know what difference it makes by moving sort?

Reply:
There are no random shapes in test_shuffle so its runtime, even give the sort, should be fairly consistent.  My only explanation for the runtime variation is that maybe there's multiple cpu runners on the same machine.  Looking for confirmation from @marcoabreu.

Moving the sort was just a style preference, not performance driven.  The biggest runtime savings came from introducing the `if stride > 1:` clause and avoiding needless work for the last big 1D test input.  The rest was just some further perf polishing.

I keep stumbling on flakey tests with this PR, so my last commit is fixing laop_6.  A bit of bad luck I would say.

Reply:
we run up to 4 jobs per windows-cpu machine. 1 job per windows-gpu machine.